lib/include/jxl/encode.h

```diff
@@ -669,47 +669,123 @@ JXL_EXPORT JxlEncoderStatus JxlEncoderAddImageFrame(
     const JxlPixelFormat* pixel_format, const void* buffer, size_t size);
 
 /**
- * TODO(firsching): add documentation
+ * The JxlEncoderOutputProcessor structure provides an interface for the
+ * encoder's output processing. Users of the library, who want to do streaming
+ * encoding, should implement the required callbacks for buffering, writing,
+ * seeking (if supported), and watermarking during the encoding process.
  *
- */
-typedef void (*JxlEncoderOutputCallback)(void* run_opaque, size_t pos,
-                                         size_t num_bytes);
-
-/**
- * TODO(firsching): add documentation
+ * At a high level, the processor can be in one of two states:
+ * - With an active buffer: This indicates that a buffer has been acquired using
+ *   `get_buffer` and encoded data can be written to it.
+ * - Without an active buffer: In this state, no data can be written. A new
+ * buffer must be acquired after releasing any previously active buffer.
+ *
+ * It is not allowed to acquire more than one buffer at a given time.
+ *
+ * The JxlEncoder OutputProcessor interacts with `position` and `watermark`,
+ * which have the following meaning.
+ *
+ * - position: Represents the current position, in bytes, within the output
+ * stream where the encoded data will be written next. This position moves
+ * forward as data is written, and can also be adjusted through the optional
+ * seek callback, if provided. At this position the next write will occur.
  *
+ * - watermark:  A position in the output stream that ensures all bytes before
+ * this point are finalized and won't be changed by later writes.
+ *
+ * All fields but `seek` are required, `seek` is optional and can be NULL.
  */
-JXL_EXPORT JxlEncoderStatus
-JxlEncoderSetOutputCallback(JxlEncoderOutputCallback callback);
+struct JxlEncoderOutputProcessor {
+  /**
+   * Required.
+   * An opaque pointer that the client can use to store custom data.
+   * This data will be passed to the associated callback functions.
+   */
+  void* opaque;
+
+  /**
+   * Required.
+   * Acquires a buffer into which the library will write the output data.
```

Comment:
Acquires a buffer at the current position ...

Reply:
done